src/app/source-file-tabs/source-file-tabs.service.ts

```diff
@@ -128,4 +131,19 @@ export class SourceFileTabsService {
     const id = ga4ghWorkflowIdPrefix + fileId;
     return this.fileService.getDownloadFilePath(id, versionName, type, relativePath);
   }
+  getPrimaryPath(
+    path: string,
+    versionName: string,
+    descriptorType: ToolDescriptor.TypeEnum,
+    descriptorPath: string,
+    entryType: EntryType
+  ): string[] {
+    const primaryPath = [];
+    primaryPath.push(this.infoTabService.getPrimaryPath(path, versionName, descriptorType, descriptorPath, entryType));
+    if (descriptorType === 'NFL') {
```

Comment:
An observation (no immediate action needed): in the future, if we want to mark certain files as "additional" primaries (which seems kinda weird but also sorta makes sense), such as the main script in a Nextflow workflow, it might be easier to mark them as such in the webservice.

Reply:
Created a new ticket: https://github.com/dockstore/dockstore/issues/5299